log4j2-logstash-reactor/src/main/java/se/fnord/logtags/log4j2_logstash/reactor/TapLogHandler.java

```diff
@@ -0,0 +1,79 @@
+package se.fnord.logtags.log4j2_logstash.reactor;
+
+import se.fnord.logtags.tags.TagLogger;
+import se.fnord.logtags.tags.Tags;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+public record TapLogHandler<T>(TagLogger logger, ValueTagLogger<T> onNext, ValueTagLogger<Throwable> onError, NoValueTagLogger onComplete, NoValueTagLogger onCancel) {
+  public static <T> TapLogHandler<T> forLogger(TagLogger logger) {
+    return new TapLogHandler<>(logger,
+        new ValueTagLogger<>(TagLogger.Level.INFO, null),
+        new ValueTagLogger<>(TagLogger.Level.ERROR, null),
+        new NoValueTagLogger(TagLogger.Level.ERROR, null),
+        new NoValueTagLogger(TagLogger.Level.INFO, null));
```

Comment:
I believe you need to swap last two, making INFO on complete and ERROR on cancel. At the same time it doesn't matter very much, because I think it's impossible to log anything without providing log level with current API state.